src/sagemaker/predictor.py

```diff
@@ -105,11 +105,27 @@ def _create_request_args(self, data, initial_args=None):
         args['Body'] = data
         return args
 
-    def delete_endpoint(self):
-        """Delete the Amazon SageMaker endpoint backing this predictor.
+    def _delete_endpoint_config(self):
+        """Delete the Amazon SageMaker endpoint configuration
+        """
+        endpoint_description = self.sagemaker_session.sagemaker_client.describe_endpoint(EndpointName=self.endpoint)
+        endpoint_config_name = endpoint_description['EndpointConfigName']
+        self.sagemaker_session.delete_endpoint_config(endpoint_config_name)
+
+    def delete_endpoint(self, delete_endpoint_config=True):
+        """Delete the Amazon SageMaker endpoint backing this predictor. Also delete the endpoint configuration attached
+           to it if delete_endpoint_config is True.
+
+        Args:
+            delete_endpoint_config (bool): Flag to indicate whether to delete endpoint configuration together with
+                endpoint. If False, only endpoint will be deleted. Default: True.
+
         """
         self.sagemaker_session.delete_endpoint(self.endpoint)
 
+        if delete_endpoint_config:
+            self._delete_endpoint_config()
```

Comment:
I think I'm a little worried about the endpoint configuration happening after the delete_endpoint, as we call ``describe_endpoint`` within the newly introduced ``_delete_endpoint_config`` to get metadata on the endpoint to determine the corresponding endpoint configuration.

I think it might be possible for the endpoint to be deleted and lose the trace to the AWS account. This might end with not being able to get the metadata needed to properly delete the endpoint config.

Reply:
You don't have to do a try-except within the _delete_endpoint_config. My main concern was that the code was doing a [delete_endpoint -> delete_endpoint_configuration](https://github.com/aws/sagemaker-python-sdk/blob/7f33ac79cd4b10f58eafd7a1b2f9af0dc5c48edd/src/sagemaker/predictor.py#L124), when the delete_endpoint_configuration calls describe_endpoint, which depends on the endpoint.

As long as we delete endpoint_configuration first and then endpoint, I think we should be fine and don't need to have extra try catch logic within the _delete_endpoint_config.